File: java/javaapi/src/main/java/io/joynr/endpoints/JoynrMessagingEndpointAddress.java
package io.joynr.endpoints;

/*
 * #%L
 * joynr::java::javaapi
 * %%
 * Copyright (C) 2011 - 2013 BMW Car IT GmbH
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *      http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */

import java.io.Serializable;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class JoynrMessagingEndpointAddress implements EndpointAddressBase, Serializable {

    /**
     * 
     */
    private static final long serialVersionUID = 2258977417169018529L;
    private String channelId;

    @JsonCreator
    public JoynrMessagingEndpointAddress(@JsonProperty("channelId") String channelId) {
        this.setChannelId(channelId);
    }

    public String getChannelId() {
        return channelId;
    }

    public void setChannelId(String channelId) {
        this.channelId = channelId;
    }

    @Override
    public int hashCode() {
        final int prime = 31;
        int result = 1;
        result = prime * result + ((channelId == null) ? 0 : channelId.hashCode());
        return result;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        JoynrMessagingEndpointAddress other = (JoynrMessagingEndpointAddress) obj;
        if (channelId == null) {
            if (other.channelId != null) {
                return false;
            }
        } else if (!channelId.equals(other.channelId)) {
            return false;
        }
        return true;
    }

    @Override
    public String toString() {
        StringBuilder stringBuilder = new StringBuilder();
        stringBuilder.append(this.getClass().getSimpleName());
        stringBuilder.append(" channel:");
        stringBuilder.append(channelId);
        return stringBuilder.toString();
    }

}

File: cpp/common/joynr-messaging/JoynrMessagingEndpointAddress.cpp
/*
 * #%L
 * %%
 * Copyright (C) 2011 - 2013 BMW Car IT GmbH
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *      http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
#include "joynr/JoynrMessagingEndpointAddress.h"

namespace joynr {

const QString& JoynrMessagingEndpointAddress::ENDPOINT_ADDRESS_TYPE() {
    static QString value("joynr::JoynrMessagingEndpointAddress");
    return value;
}

JoynrMessagingEndpointAddress::JoynrMessagingEndpointAddress() :
    EndpointAddressBase(),
    channelId()
{

}

JoynrMessagingEndpointAddress::JoynrMessagingEndpointAddress(const QString &channelId) :
    EndpointAddressBase(),
    channelId(channelId)
{

}

JoynrMessagingEndpointAddress::JoynrMessagingEndpointAddress(const JoynrMessagingEndpointAddress &other) :
    EndpointAddressBase(other),
    channelId(other.getChannelId())
{
}

JoynrMessagingEndpointAddress::~JoynrMessagingEndpointAddress(){

}

JoynrMessagingEndpointAddress &JoynrMessagingEndpointAddress::operator =(const JoynrMessagingEndpointAddress &other)
{
    EndpointAddressBase::operator =(other);
    channelId = other.getChannelId();
    return *this;
}

bool JoynrMessagingEndpointAddress::operator ==(const JoynrMessagingEndpointAddress &other) const
{
    return EndpointAddressBase::operator ==(other)
            && channelId == other.getChannelId();
}

void JoynrMessagingEndpointAddress::setChannelId(const QString &channelId){
    this->channelId = channelId;
}

QString JoynrMessagingEndpointAddress::getChannelId() const{
    return channelId;
}



} // namespace joynr
